shared/rocroller/test/catch/ControlFlowRWTracerTest.cpp

```diff
@@ -0,0 +1,1086 @@
+    auto                             records = tracer.coordinatesReadWrite();
+
+    // SetCoordinate doesn't read/write
+    CHECK(records.size() == 0);
+}
+
+TEST_CASE("ControlFlowRWTracer - Assign Read and Write", "[kernel-graph][utils]")
+{
+    KernelGraph::KernelGraph graph;
+
+    auto tileA = graph.coordinates.addElement(MacroTile());
+    auto tileB = graph.coordinates.addElement(MacroTile());
+    auto tileC = graph.coordinates.addElement(MacroTile());
+
+    auto kernel = graph.control.addElement(Kernel());
+
+    auto exprA = dataFlowTag(tileA, Register::Type::Vector, DataType::Float);
+    auto exprB = dataFlowTag(tileB, Register::Type::Vector, DataType::Float);
+
+    auto assign = graph.control.addElement(Assign{Register::Type::Vector, exprA + exprB});
+
+    graph.control.addElement(Body(), {kernel}, {assign});
+    graph.mapper.connect(assign, tileC, NaryArgument::DEST);
+
+    KernelGraph::ControlFlowRWTracer tracer(graph);
+
+    // Check that tileA and tileB are READ
+    auto tileARecords = tracer.coordinatesReadWrite(tileA);
+    REQUIRE(tileARecords.size() == 1);
+    CHECK(tileARecords[0].control == assign);
+    CHECK(tileARecords[0].coordinate == tileA);
+    CHECK(tileARecords[0].rw == RW::READ);
+
+    auto tileBRecords = tracer.coordinatesReadWrite(tileB);
+    REQUIRE(tileBRecords.size() == 1);
+    CHECK(tileBRecords[0].control == assign);
+    CHECK(tileBRecords[0].coordinate == tileB);
+    CHECK(tileBRecords[0].rw == RW::READ);
+
+    // Check that tileC is WRITE
+    auto tileCRecords = tracer.coordinatesReadWrite(tileC);
+    REQUIRE(tileCRecords.size() == 1);
+    CHECK(tileCRecords[0].control == assign);
+    CHECK(tileCRecords[0].coordinate == tileC);
+    CHECK(tileCRecords[0].rw == RW::WRITE);
+}
+
+TEST_CASE("ControlFlowRWTracer - LoadTiled and StoreTiled", "[kernel-graph][utils]")
+{
+    KernelGraph::KernelGraph graph;
+
+    auto tileA = graph.coordinates.addElement(MacroTile());
+    auto tileB = graph.coordinates.addElement(MacroTile());
+    auto userA = graph.coordinates.addElement(User());
+    auto userB = graph.coordinates.addElement(User());
+
+    auto kernel = graph.control.addElement(Kernel());
+    auto load   = graph.control.addElement(LoadTiled());
+    auto store  = graph.control.addElement(StoreTiled());
+
+    graph.control.addElement(Body(), {kernel}, {load});
+    graph.control.addElement(Sequence(), {load}, {store});
+
+    graph.mapper.connect<MacroTile>(load, tileA);
+    graph.mapper.connect<User>(load, userA);
+    graph.mapper.connect<MacroTile>(store, tileB);
+    graph.mapper.connect<User>(store, userB);
+
+    KernelGraph::ControlFlowRWTracer tracer(graph);
+
+    // LoadTiled should write to tileA
+    auto tileARecords = tracer.coordinatesReadWrite(tileA);
+    bool foundWrite   = false;
+    for(auto const& record : tileARecords)
+    {
+        if(record.control == load && record.rw == RW::WRITE)
+        {
+            foundWrite = true;
+        }
+    }
```

Comment:
Minor: break when found.
Maybe:
```suggestion
    bool foundWrite  =
        std::ranges::any_of(tileARecords,
                            [load](const auto& x) {
                               return x.control == load and x.rw == RW::WRITE
                            });
```

Reply:
Done throughout